test/cuNDArray_blas_test.cpp

```diff
@@ -7,150 +7,235 @@
-    Array = cuNDArray<T>(&dims);
-    Array2 = cuNDArray<T>(&dims);
-  }
-  std::vector<size_t> dims;
-  cuNDArray<T> Array;
-  cuNDArray<T> Array2;
+template <typename T> class cuNDArray_blas_Real : public ::testing::Test {
+  protected:
+    virtual void SetUp() {
+        size_t vdims[] = {37, 49, 23, 19}; // Using prime numbers for setup because they are messy
+
+        dims = std::vector<size_t>(vdims, vdims + sizeof(vdims) / sizeof(size_t));
+
+        // test updated code
+        size_t vdims_large[] = {size_t(INT_MAX) + size_t(10000)}; // Using a large vector to test updated code
+        dims_large = std::vector<size_t>(vdims_large, vdims_large + sizeof(vdims_large) / sizeof(size_t));
+
+        Array = cuNDArray<T>(&dims);
+        Array2 = cuNDArray<T>(&dims);
+        Array3 = cuNDArray<T>(&dims_large);
+        Array4 = cuNDArray<T>(&dims_large);
+    }
+    std::vector<size_t> dims;
+    std::vector<size_t> dims_large;
+    cuNDArray<T> Array;
+    cuNDArray<T> Array2;
+    cuNDArray<T> Array3;
+    cuNDArray<T> Array4;
 };
 
 typedef Types<float, double> realImplementations;
 
 TYPED_TEST_CASE(cuNDArray_blas_Real, realImplementations);
 
-TYPED_TEST(cuNDArray_blas_Real,dotTest){
-  fill(&this->Array,TypeParam(1));
-  EXPECT_FLOAT_EQ(this->Array.get_number_of_elements(),real(dot(&this->Array,&this->Array)));
-  fill(&this->Array2,TypeParam(2));
-  EXPECT_FLOAT_EQ(this->Array.get_number_of_elements()*2,real(dot(&this->Array,&this->Array2)));
-}
+TYPED_TEST(cuNDArray_blas_Real, dotTest) {
+    fill(&this->Array, TypeParam(1));
+    EXPECT_FLOAT_EQ(this->Array.get_number_of_elements(), real(dot(&this->Array, &this->Array)));
+    fill(&this->Array2, TypeParam(2));
+    EXPECT_FLOAT_EQ(this->Array.get_number_of_elements() * 2, real(dot(&this->Array, &this->Array2)));
 
-TYPED_TEST(cuNDArray_blas_Real,axpyTest){
-  fill(&this->Array,TypeParam(71));
-  fill(&this->Array2,TypeParam(97));
-  axpy(TypeParam(11),&this->Array,&this->Array2);
-  TypeParam val = this->Array2[10];
-  EXPECT_FLOAT_EQ(878,real(val));
+    fill(&this->Array3, TypeParam(1));
+    EXPECT_FLOAT_EQ(this->Array3.get_number_of_elements(), real(dot(&this->Array3, &this->Array3)));
+    fill(&this->Array4, TypeParam(2));
+    EXPECT_FLOAT_EQ(this->Array3.get_number_of_elements() * 2, real(dot(&this->Array3, &this->Array4)));
 }
 
-TYPED_TEST(cuNDArray_blas_Real,nrm2Test){
-  fill(&this->Array,TypeParam(1));
-  EXPECT_FLOAT_EQ(std::sqrt((double)this->Array.get_number_of_elements()),nrm2(&this->Array));
-  fill(&this->Array,TypeParam(3));
-  EXPECT_FLOAT_EQ(std::sqrt(3.0*3.0*this->Array.get_number_of_elements()),nrm2(&this->Array));
+TYPED_TEST(cuNDArray_blas_Real, axpyTest) {
+    fill(&this->Array, TypeParam(71));
+    fill(&this->Array2, TypeParam(97));
+    axpy(TypeParam(11), &this->Array, &this->Array2);
+    TypeParam val = this->Array2[10];
+    EXPECT_FLOAT_EQ(878, real(val));
+
+    fill(&this->Array3, TypeParam(71));
```

Comment:
I think if the goal here is to test that batched versions of these functions work, we should probably test values beyond the first batch? Or some aggregate?